gui/widgets/qopenglmousetracker.cpp

```diff
@@ -20,6 +20,66 @@ QOpenGLMouseTracker::QOpenGLMouseTracker(QWidget *parent, float lineWidth,
       m_backgroundColor(QWidget::palette().color(QWidget::backgroundRole())),
       m_lastColour(QWidget::palette().color(QWidget::backgroundRole()).rgb()) {}
 
+std::shared_ptr<rendering::ClippingPlane>
+QOpenGLMouseTracker::BuildClippingPlane(GLfloat a, GLfloat b, GLfloat c,
+                                        GLfloat d, bool active) {
+
+  auto it = m_clippingPlanesPool.begin();
+  assert(it != m_clippingPlanesPool.end());
```

Comment:
if this could happen then it should be a run-time check - in a release build the assert does nothing and if the pool is empty the next line segfaults